packages/zosfiles/src/methods/upload/Upload.ts

```diff
@@ -854,37 +854,33 @@ export class Upload {
         };
     }
 
-    public static async uploadFile(session: AbstractSession, localPath: string, ussPath: string,
-        options: IUploadOptions): Promise<IZosFilesResponse> {
-        const tempOptions: Partial<IUploadOptions> = {};
-
+    public static async uploadFile(
+        session: AbstractSession,
+        localPath: string,
+        ussPath: string,
+        options: IUploadOptions
+    ): Promise<IZosFilesResponse> {
         if (options.attributes) {
             if (!options.attributes.fileShouldBeUploaded(localPath)) {
                 return;
             }
-            tempOptions.binary = options.attributes.getFileTransferMode(localPath, options.binary) === TransferMode.BINARY;
+            options.binary = options.attributes.getFileTransferMode(localPath, options.binary) === TransferMode.BINARY;
```

Comment:
Didn't we create tempOptions because this gets called many times when recursively uploading a USS directory, and changing the options object changes the options for other uploads in progress when concurrent requests are being used?